tools/js-sdk-release-tools/src/test/changelog/automaticGenerateChangeLogAndBumpVersion.test.ts

```diff
@@ -0,0 +1,249 @@
+        }
+        vi.clearAllMocks();
+        vi.resetModules();
+    });
+
+    test('should call getApiReviewPath for both npm and local packages when processing existing track2 SDK', async () => {
+        // Setup: Create HLC package with -node.api.md file
+        const packageJson = {
+            name: '@azure/arm-testservice',
+            version: '2.0.0',  // Changed to 2.0.0 to simulate existing package
+            'sdk-type': 'mgmt'
+        };
+        
+        await writeFile(
+            path.join(packageDir, 'package.json'), 
+            JSON.stringify(packageJson, null, 2)
+        );
+
+        // Create both files, but -node.api.md should take priority
+        const nodeApiContent = `
+// Test API content for Node.js specific API
+export interface TestServiceClient {
+    // Node.js specific methods
+    testOperation(): Promise<string>;
+}
+`;
+
+        const standardApiContent = `
+// Standard API content
+export interface TestServiceClient {
+    // Standard methods
+    standardOperation(): Promise<string>;
+}
+`;
+
+        await writeFile(
+            path.join(packageDir, 'review', 'arm-testservice-node.api.md'),
+            nodeApiContent
+        );
+        
+        await writeFile(
+            path.join(packageDir, 'review', 'arm-testservice.api.md'),
+            standardApiContent
+        );
+
+        // Create a basic TypeScript client file
+        const clientContent = `
+export class TestServiceClient {
+    testOperation(): Promise<string> {
+        return Promise.resolve("test");
+    }
+}
+`;
+        await writeFile(path.join(packageDir, 'src', 'testServiceClient.ts'), clientContent);
+
+        // Create changelog-temp directory structure for npm package comparison
+        const changelogTempDir = path.join(packageDir, 'changelog-temp', 'package');
+        await mkdir(changelogTempDir, { recursive: true });
+        await mkdir(path.join(changelogTempDir, 'review'), { recursive: true });
+        
+        // Create npm package.json with sdk-type: mgmt
+        const npmPackageJson = {
+            name: '@azure/arm-testservice',
+            version: '1.2.0',
+            'sdk-type': 'mgmt'
+        };
+        await writeFile(
+            path.join(changelogTempDir, 'package.json'),
+            JSON.stringify(npmPackageJson, null, 2)
+        );
+
+        // Create npm package API review file
+        await writeFile(
+            path.join(changelogTempDir, 'review', 'arm-testservice-node.api.md'),
+            `// Old API content\nexport interface TestServiceClient {\n    oldOperation(): Promise<string>;\n}`
+        );
+
+        // Create npm package CHANGELOG.md
+        await writeFile(
+            path.join(changelogTempDir, 'CHANGELOG.md'),
+            `# Release History\n\n## 1.2.0 (2023-03-01)\n\n### Features\n\n- Previous feature`
+        );
+
+        // Mock shell.pwd() using dynamic import approach
+        vi.doMock('shelljs', () => ({
```

Comment:
I'm not too familiar with this codebase, but generally I would recommend refactoring any shell calls out to their own functions and mocking those as opposed to mocking external packages directly.